test/e2e/kb/stack_monitoring_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build kb e2e
+
+package kb
+
+import (
+	"testing"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/stackmon/validations"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/checks"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/kibana"
+)
+
+// TestKBStackMonitoring tests that when a Kibana is configured with monitoring, its log and metrics are
+// correctly delivered to the referenced monitoring Elasticsearch clusters.
+func TestKBStackMonitoring(t *testing.T) {
+	// only execute this test on supported version
+	err := validations.IsSupportedVersion(test.Ctx().ElasticStackVersion)
+	if err != nil {
+		t.SkipNow()
+	}
+
+	// create 1 monitored and 2 monitoring clusters to collect separately metrics and logs
+	metrics := elasticsearch.NewBuilder("test-kb-mon-metrics").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	logs := elasticsearch.NewBuilder("test-kb-mon-logs").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	assocEs := elasticsearch.NewBuilder("test-kb-mon-a").
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources)
+	monitored := kibana.NewBuilder("test-kb-mon-a").
+		WithElasticsearchRef(assocEs.Ref()).
+		WithNodeCount(1).
+		WithMonitoring(metrics.Ref(), logs.Ref())
+
+	// checks that the sidecar beats have sent data in the monitoring clusters
+	steps := func(k *test.K8sClient) test.StepList {
+		c := checks.StackMonitoringChecks{
+			MonitoredNsn: k8s.ExtractNamespacedName(&monitored.Kibana),
+			Metrics:      metrics,
+			Logs:         logs,
+			K:            k,
+		}
+		return test.StepList{
+			c.CheckBeatSidecars(),
+			//c.CheckMetricbeatIndex(), TODO: investigate if it's normal that there is no document in this index when es monitoring is off
```

Comment:
See my other comment, I'm not sure to understand why enabling es monitoring would move the stack metrics from `.monitoring-*` to `metricbeat-*`. I'll try to investigate.

Reply:
If I enable Elasticsearch monitoring I get the expected `.monitoring-es-` index, but there is also a `metricbeat-` index created:

```
.monitoring-kibana-7-mb-2021.07.12  0     p      STARTED  142 183.4kb 10.55.210.25 monitoring-es-default-1
.monitoring-kibana-7-mb-2021.07.12  0     r      STARTED  142   166kb 10.55.209.29 monitoring-es-default-2
.monitoring-es-7-mb-2021.07.12      0     p      STARTED  839     1mb 10.55.210.25 monitoring-es-default-1
.monitoring-es-7-mb-2021.07.12      0     r      STARTED  195 682.8kb 10.55.208.23 monitoring-es-default-0
metricbeat-7.14.0-2021.07.12-000001 0     r      STARTED   24 151.6kb 10.55.208.23 monitoring-es-default-0
metricbeat-7.14.0-2021.07.12-000001 0     p      STARTED   24 151.6kb 10.55.209.29 monitoring-es-default-2
```

Having a look at these 24 documents in `metricbeat-7.14.0-2021.07.12-000001` they are actually all errors which, I guess, happened during metricbeat startup:

| Time                        | _index                              | error.message                                                                                                                                                                                  |
|-----------------------------|-------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Jul 12, 2021 @ 13:28:09.677 | metricbeat-7.14.0-2021.07.12-000001 | error determining if connected Elasticsearch node is master: error making http request: Get "https://localhost:9200/_nodes/_local/nodes": dial tcp 127.0.0.1:9200: connect: connection refused |

I think `metricbeat-*` should not be used in e2e tests to assess that monitoring is working as expected.

Reply:
Fixed in https://github.com/elastic/cloud-on-k8s/pull/4618/commits/b600503d7787fe374fea10f07a87d065d0c12856